### PR TITLE
Redux

### PR DESCRIPTION
**Pull request by mjoyce91**
This PR includes changes to introduce Redux into the application. Connected components (using `connect` from react-redux) are used in order to provide presentational components with data. Reducers and actions were created to retrieve filters, position details, posts, and search results.

### CODE REVIEW

File: src/Components/PostDetails/PostDetails.jsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    );
+    return (
+      <div>
+        {postMissionData}
+      </div>
+    );
+  }
+}
+
+PostDetails.propTypes = {
```
Comment (burgwyn) [inline src/Components/PostDetails/PostDetails.jsx]: Do we want to externalize these proptypes?

File: src/Constants/PropTypes.js [hunk 2]
```diff
@@ -25,12 +25,12 @@ export const POST_MISSION_DATA = PropTypes.shape({
 });
 
 export const DETAILS = PropTypes.shape({
```
Comment (burgwyn) [inline src/Constants/PropTypes.js]: Let's rename to a more specific kind of `DETAILS`

File: src/Containers/Home/Home.test.jsx [hunk 3]
```diff
@@ -1,128 +1,24 @@
...
-          { id: 3, code: 'AB', long_description: 'Albanian', short_description: 'Albanian', effective_date: '1984-05-04' },
-          { id: 4, code: 'AC', long_description: 'Amharic', short_description: 'Amharic', effective_date: '1984-05-04' },
-    ],
-    [
-          { id: 2, code: '00' },
-          { id: 3, code: '01' },
-    ],
-  ];
+describe('Main', () => {
+  const api = 'http://localhost:8000/api/v1';
```
Comment (burgwyn) [inline src/Containers/Home/Home.test.jsx]: Is it possible to change the host name/port?  Can we use something that references this is a unit test?

File: src/Components/PostDetails/PostDetails.jsx (hunk 1 above)
Comment (mjoyce91) [inline src/Components/PostDetails/PostDetails.jsx]: Good catch - missed this one.

Comment (mjoyce91) [inline src/Components/PostDetails/PostDetails.jsx]: Resolved with 47b4bac

File: src/Constants/PropTypes.js (hunk 2 above)
Comment (mjoyce91) [inline src/Constants/PropTypes.js]: Updated with `POSITION_DETAILS`

File: src/Containers/Home/Home.test.jsx (hunk 3 above)
Comment (burgwyn) [inline src/Containers/Home/Home.test.jsx]: That's an idea.  The scenario I'm thinking of is `ctl+F` on 'localhost' and seeing test files in the mix.  We can table this for later.

**Pull request merged**